### DISCUSSION TITLE
Dynamic routing + dynamic tab + sub route

### DISCUSSION THREAD

**Discussion by nijeshw** [hannoeru/vite-plugin-pages#433] [Q&A]
### Describe the bug

How to handle sub route after dynamic tab

`http://localhost:5173/clients/view/1/pre-assessment` --> `http://localhost:5173/clients/view/[id]/[tab]`

This above URL is working file, I would like to add sub route(`list`, `add`) after `pre-assessment` [tab] so my new URL should like 
`http://localhost:5173/clients/view/1/pre-assessment/list`
`http://localhost:5173/clients/view/1/pre-assessment/add`
The below is the file structure that I have followed now

![image](https://github.com/hannoeru/vite-plugin-pages/assets/2499917/9ccfb289-d305-44f5-8f58-f701dec5da63)

can anyone help me to make this sub route.

Thanks

### Reproduction

nill

### System Info

```shell
I'm using vue 3.2.47 and vite-plugin-pages 0.29.0
```

### Used Package Manager

yarn

### Validations

- [X] Follow the [Code of Conduct](https://github.com/hannoeru/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the docs.
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A please open a GitHub Discussion.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by hannoeru** [comment]
Just add `list.vue` and `add.vue` file under `[tab]` folder.

**Reply by nijeshw** [reply]
Thanks for your message, I have multiple tabs which should also need sub route `list` and `add` so it will conflict with previous route right ?

`http://localhost:5173/clients/view/1/pre-assessment`
`http://localhost:5173/clients/view/1/risk-assessment`
`http://localhost:5173/clients/view/1/carer-preference`

this three tabs I have sub route `list` and `add` so the full URL should be like below

`http://localhost:5173/clients/view/1/pre-assessment/list`
`http://localhost:5173/clients/view/1/pre-assessment/add`

`http://localhost:5173/clients/view/1/risk-assessment/list`
`http://localhost:5173/clients/view/1/risk-assessment/add`

`http://localhost:5173/clients/view/1/carer-preference/list`
`http://localhost:5173/clients/view/1/carer-preference/add`

**Reply by hannoeru** [reply]
I didn't get it but I think this should works

Option 1

`/clients/view/[id]/pre-assessment/index.vue`
`/clients/view/[id]/risk-assessment/index.vue`
`/clients/view/[id]/carer-preference/index.vue`
`/clients/view/[id]/[tab]/list.vue`
`/clients/view/[id]/[tab]/add.vue`

Option 2

`/clients/view/[id]/[tab]/index.vue` (Use switch to display different view base on tab
`/clients/view/[id]/[tab]/list.vue`
`/clients/view/[id]/[tab]/add.vue`
